### PR TITLE
Update sbt to 1.9.0, migrate to new sbt syntax && Modernize a bit the CI (add ScalaSteward, dependabot and release-drafter)

### LINKED ISSUE #219: Upgrade project/build.properties to use SBT 1.9.0
Now that SBT 1.9.0 is released, would it be possible to release this plugin with this version of SBT? This would make it available to thousands of developers who are desperate to benefit from easy performance benchmarking in Scala & Java with SBT :-)

Thanks so much!

### PR DESCRIPTION
**Pull request by guizmaii**
Fixes #219

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,11 +14,13 @@ jobs:
         include:
         - java: 8
         - java: 11
+        - java: 17
     steps:
     - uses: actions/checkout@v2.3.4
-    - uses: olafurpg/setup-scala@v10
```
Comment (guizmaii) [inline .github/workflows/ci.yml:19]: Replace by `setup-java` as `setup-scala` is using Jabba, which is not maintained anymore

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+version: 2
+updates:
+  - package-ecosystem: "github-actions"
```
Comment (guizmaii) [inline .github/dependabot.yml:3]: Helps maintaining Github Actions workflows up-to-date

File: .github/workflows/release-drafter.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+on:
+  push:
+    branches: ['master']
+
+jobs:
+  update_release_draft:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: release-drafter/release-drafter@v5
```
Comment (guizmaii) [inline .github/workflows/release-drafter.yml:11]: Automatically maintain a changelog in the Github releases

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mzuehlke) [reply .github/workflows/ci.yml:19]: `Setup-java` includes caching:
`cache: sbt`
Then you could remove the `rm` command, too

File: .github/workflows/scala-steward.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+name: Scala Steward
```
Comment (mzuehlke) [inline .github/workflows/scala-steward.yml:1]: Other repos in this organisation are using the public Scala-steward, by adding the repo here: https://github.com/search?q=repo%3AVirtusLab%2Fscala-steward-repos%20sbt%2Fsbt-&type=code

Comment (guizmaii) [reply .github/workflows/scala-steward.yml:1]: IMO, it's better to use the Github Actions. It's local to the repo, you have more control to when it's launched, you can launch it manually if needed and the PRs don't come from a fork so they're easier to update if necessary

**Comment by guizmaii** [comment]
@mzuehlke 
> Then you could remove the rm command, too

Removed

> Setup-java includes caching:
> cache: sbt

We're already using `coursier/cache-action` for the cache

File: .github/workflows/scala-steward.yml (hunk 4 above)
Comment (mzuehlke) [reply .github/workflows/scala-steward.yml:1]: Makes sense

**Comment by guizmaii** [comment]
@mzuehlke Fixed!

**Review by johanandren — APPROVED** [review]
LGTM, thanks. I'll see about a 0.4.5 release.

**Pull request merged**